adapters/ix/ix.go

```diff
@@ -99,29 +82,35 @@ func (a *IxAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pbs.P
 		}
 
 		for sizeIndex, format := range unit.Sizes {
-			// Only grab this ad unit
-			// Not supporting multi-media-type adunit yet
+			// Only grab this ad unit. Not supporting multi-media-type adunit yet.
 			thisImp := indexReqImp[i]
 
 			thisImp.TagID = unit.Code
-			thisImp.Banner.Format = []openrtb.Format{format}
-			thisImp.Banner.W = &format.W
-			thisImp.Banner.H = &format.H
+			if thisImp.Banner != nil {
+				thisImp.Banner.Format = []openrtb.Format{format}
```

Comment:
Thank you for adding the nil check. Although we plan to remove the legacy code later this year, it's nice to fix the "broken windows" until then. In our newer adapters we would ask for the banner object to be shallow copied first due to pointer sharing, but this object is created by `MakeOpenRTBGeneric` earlier in this adapter so I expect we're fine.